decidim-core/spec/system/last_activity_spec.rb

```diff
@@ -121,9 +121,9 @@
 
       context "when there are activities from private spaces" do
         before do
-          component.participatory_space.update(private_space: true)
-          comment.participatory_space.update(private_space: true)
-          another_comment.participatory_space.update(private_space: true)
+          Decidim::ActionLog.find_each do |action_log|
+            action_log.participatory_space.update(private_space: true)
+          end
```

Comment:
Could this be ignoring a bug where we're leaking comments of private resources? If I leave the older code:

```ruby
          component.participatory_space.update(private_space: true)
          comment.participatory_space.update(private_space: true)
          another_comment.participatory_space.update(private_space: true)
```

Then I have a failing spec. Because it's finding these comments. Or am I missing something?

Reply:
I think you miss this part, where private stuff are being filtered out: 

```
def activities
  @activities ||= last_activities.select do |activity|
    activity.visible_for?(current_user)
  end
end
```

Reply:
As I just commented, @alecslupu, that visible_for, as far as I understand has been moved to the query layer, instead of filtering in the cell

Reply:
@alecslupu I think it's more difficult to actually see it with the redesign changes, but applying this patch with the current PR you can see what I mean: 

```diff
diff --git a/decidim-core/app/cells/decidim/activity/show.erb b/decidim-core/app/cells/decidim/activity/show.erb
index a9244338b4..7fd174d246 100644
--- a/decidim-core/app/cells/decidim/activity/show.erb
+++ b/decidim-core/app/cells/decidim/activity/show.erb
@@ -8,6 +8,8 @@
       <a href="<%= resource_link_path %>">
         <%= html_truncate decidim_sanitize(resource_link_text, strip_tags: true), length: 80 %>
       </a>
+
+      <%= comment.presence if defined? comment%>
     </span>
     <span>
       <%= icon "treasure-map-line", class: "w-3.5 h-3.5 text-gray fill-current" %>
diff --git a/decidim-core/spec/system/last_activity_spec.rb b/decidim-core/spec/system/last_activity_spec.rb
index 410f2bb775..89cf02834b 100644
--- a/decidim-core/spec/system/last_activity_spec.rb
+++ b/decidim-core/spec/system/last_activity_spec.rb
@@ -121,9 +121,13 @@ describe "Last activity", type: :system do
 
       context "when there are activities from private spaces" do
         before do
-          Decidim::ActionLog.find_each do |action_log|
-            action_log.participatory_space.update(private_space: true)
-          end
+          comment.update(body: { es: "this is a private comment" })
+          another_comment.update(body: { es: "this is another private comment" })
+
+          component.participatory_space.update(private_space: true)
+          comment.participatory_space.update(private_space: true)
+          another_comment.participatory_space.update(private_space: true)
+
           visit current_path
         end
```

By running this I see the following failed expectation: 

```shell
bin/rspec decidim-core/spec/system/last_activity_spec.rb:122
Run options: include {:locations=>{"./spec/system/last_activity_spec.rb"=>[122]}}

Randomized with seed 40435

Last activity
  accessing the homepage with the last activity content block enabled
    when viewing all activities
      when there are activities from private spaces
        does not show the activities (FAILED - 1)

Failures:

  1) Last activity accessing the homepage with the last activity content block enabled when viewing all activities when there are activities from private spaces does not show the activities
     Failure/Error: expect(page).to have_css("[data-activity]", count: 0)
       expected to find visible css "[data-activity]" 0 times, found 2 matches: "1 day ago\nNew comment at Humberto Kovacek this is a private comment # treasure-map-line Eveniet dicta beatae. 64", "2 days ago\nNew comment at Jon Kemmer this is another private comment # treasure-map-line Eaque ut inventore. 187"
     
     [Screenshot Image]: file:///home/apereira/Work/decidim/decidim/spec/decidim_dummy_app/tmp/screenshots/failures_r_spec_example_groups_last_activity_accessing_the_homepage_with_the_last_activity_content_block_enabled_when_viewing_all_activities_when_there_are_activities_from_private_spaces_does_not_show_the_acti_182.png

     [Screenshot HTML]: file:///home/apereira/Work/decidim/decidim/spec/decidim_dummy_app/tmp/screenshots/failures_r_spec_example_groups_last_activity_accessing_the_homepage_with_the_last_activity_content_block_enabled_when_viewing_all_activities_when_there_are_activities_from_private_spaces_does_not_show_the_acti_182.html

     

     
     # ./spec/system/last_activity_spec.rb:135:in `block (5 levels) in <top (required)>'

Top 1 slowest examples (13.28 seconds, 91.0% of total time):
  Last activity accessing the homepage with the last activity content block enabled when viewing all activities when there are activities from private spaces does not show the activities
    13.28 seconds ./spec/system/last_activity_spec.rb:134

Finished in 14.6 seconds (files took 3.49 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/system/last_activity_spec.rb:134 # Last activity accessing the homepage with the last activity content block enabled when viewing all activities when there are activities from private spaces does not show the activities

Randomized with seed 40435

Coverage report generated for RSpec to /home/apereira/Work/decidim/decidim/coverage. 9214 / 97594 LOC (9.44%) covered.
Stopped processing SimpleCov as a previous error not related to SimpleCov has been detected
```

And this is the screenshot: 

![failures_r_spec_example_groups_last_activity_accessing_the_homepage_with_the_last_activity_content_block_enabled_when_viewing_all_activities_when_there_are_activities_from_private_spaces_does_not_show_the_acti_182](https://github.com/decidim/decidim/assets/717367/639a6be5-5ed9-45b2-b546-8b66efe79db4)

So, it seems like we're introducing a regression bug here as far as I see.

Reply:
Actually, this was a false positive. Apparently the spec itself used the resource with a certain participatory space, and the activity log used a different participatory space. resulting in a `log.resource.participatory_space != log.participatory_space`. This was actually causing this spec to fail.